### DISCUSSION TITLE
Orthographic camera

### DISCUSSION THREAD

**Discussion by rbignon** [pokepetter/ursina#704] [Q&A]
Hello,

I would like to use an orthographic camera, as my game displays a map. However, I don't understand something.

The map is composed of plane entities, and there are cube entities on.

When I start with the default `EditorCamera`, it displays:

![DeepinScreenshot_x-terminal-emulator_20240813160903](https://github.com/user-attachments/assets/651e235a-309e-48c7-a47d-a136f72db3bc)

Then, I toggle to set the orthographic camera:

![DeepinScreenshot_x-terminal-emulator_20240813161028](https://github.com/user-attachments/assets/8cd6ff6e-a8db-4ea0-aff9-8d737f91bd47)

I don't understand why, but this is truncated on the bottom.

Then, I come back on the normal mode, and unzoom to see all the map:

![DeepinScreenshot_x-terminal-emulator_20240813160929](https://github.com/user-attachments/assets/a36cb7b9-1c18-45e7-b182-fdcfe87c3a11)

And when I toggle again to orthographic mode, the output is ok:
![DeepinScreenshot_x-terminal-emulator_20240813160935](https://github.com/user-attachments/assets/4c8291b1-08a4-4ae8-bbb3-21c41cc5fd21)

Do you know why?

**Comment by pokepetter** [comment]
There's still a clip plane in orthographic. Simply move the camera further back and you'll see everything.

**Reply by rbignon** [reply]
Hm, it doesn't work:

https://www.youtube.com/watch?v=4UN9yGpIbss

Also another question: what are `camera.position`, `camera.world_position` and `EditCamera.position`? I understand than in the orthographic mode, `camera.position` is not in use and `camera.fov` is used by the zoom.

**Comment by pokepetter** [comment]
Seems like you didn't set it for enough away then. What value did you set it to? The camera is still in 3D space even the orthographic and should have enough room in front of it. Try for example: `camera.z = -500`

The EditorCamera will parent the camera to itself and take control of it. In perspective mode, it will move the camera back and forth based on this. In orthographic mode, it will change fov, in order to get a similar zooming effect for orthographic. 

`camera.position` is the local position of the camera. When parented to the editor camera, it will be at something like `Vec3(0,0,-20)`. Zooming in perspective mode will change this.

`camera.world_position` is the world position of the camera, which means the position, not relative to any parents. Chaning this when the camera is parented to the EditorCamera could make for a very strange result, since it would most likely rotate around a point that's not right in front of the camera.  

Changing `editor_camera.position` would change the position of the EditorCamera and thus the point the camera is rotating around.
